### PR TITLE
Made Builds Significantly Faster By Splitting Code Into Libraries

### PR DESCRIPTION
**Pull request by garethellis0**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [ ] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by garethellis0** [comment]
There's a circular dependency I need to resolve, but then this should bring us down to about a 18-19min CI time.

File: src/thunderbots/software/CMakeLists.txt [hunk 1]
```diff
@@ -122,25 +120,150 @@ include_directories(
...
-# Find all the generated protobuf source files
-file(GLOB PROTO_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/proto/*.pb.cc")
+###############
+## Libraries ##
+###############
 
-file(GLOB_RECURSE NETWORK_INPUT_SRC LIST_DIRECTORIES false CONFIGURE_DEPENDS
-        ${CMAKE_CURRENT_SOURCE_DIR}/network_input/*.cpp
-        ${CMAKE_CURRENT_SOURCE_DIR}/network_input/*.h
+# TODO: Should plural libraries be plurally named?
```
Comment (henryymliu) [inline src/thunderbots/software/CMakeLists.txt:127]: example?

**Review by henryymliu — DISMISSED** [review]
LGTM

File: src/thunderbots/software/CMakeLists.txt (hunk 1 above)
Comment (garethellis0) [reply src/thunderbots/software/CMakeLists.txt:127]: `tbot_primitive` or `tbot_primitives`?

**Review by MathewMacDougall — APPROVED** [review]

**Pull request merged**